### DISCUSSION TITLE
Simple automation with snappy (calling webook on new mail)

### DISCUSSION THREAD

**Discussion by shalak** [the-djmaze/snappymail#1260] [Q&A]
Hello!

I'm looking for a solution, that will perform certain actions on when new email arrives, I hope I came to the right place.

For the most basic use-case, when an email from given sender is received, I want to call a web service and POST the body of the email there. Is it possible out-of-the-box (or almost out-of-the-box) in SnappyMail?

**Comment by cm-schl** [comment]
Hi,

as SnappyMail itself is a webmail client I'm not sure how this could be achieved: Such an automatic action normally is something that you want to run autonomously without a user that logs in into SnappyMail.
What SnappyMail could do for you is give you a (actually simple) interface to a SIEVE system on your IMAP server - that is called "Filters" in SnapyyMail and every user can set them in the preferences when available on the mail server itself.
But this is maybe not that what you are looking for...

**Reply by shalak** [reply]
I was hoping to write a custom snappy plugin that will enhance the sieve-filtering feature with new "THEN" caluse which would be to call a POST webhook.

I'm looking for something that will reliably handle all the IMAP caveats (multiparts, different or missing encoding, etc) for handling the message itself
